buildsystem/spack/ascent/bsub.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+#BSUB -P csc359
+#BSUB -W 2:00
+#BSUB -nnodes 1
+#BSUB -J exasgd_spack_install
+#BSUB -o spack_install.%J
+#BSUB -e spack_install.%J
+
+exit() {
+  # Clear all trap handlers so this isn't echo'ed multiple times, potentially
+  # throwing off the CI script watching for this output
+  trap - `seq 1 31`
+
+  # If called without an argument, assume not an error
+  local ec=${1:-0}
+
+  # Echo the snippet the CI script is looking for
+  echo BUILD_STATUS:${ec}
+
+  # Actually exit with that code, although it won't matter in most cases, as CI
+  # is only looking for the string 'BUILD_STATUS:N'
+  builtin exit ${ec}
+}
+
+# This will be the catch-all trap handler after arguments are parsed.
+cleanup() {
+  # Clear all trap handlers
+  trap - `seq 1 31`
+
+  # When 'trap' is invoked, each signal handler will be a curried version of
+  # this function which has the first argument bound to the signal it's catching
+  local sig=$1
+
+  echo
+  echo Exit code $2 caught in build script triggered by signal ${sig}.
+  echo
+
+  exit $2
+}
+
+# Configure https proxy because spack is going to do some things with git
+export all_proxy="socks://proxy.ccs.ornl.gov:3128"
+export ftp_proxy="ftp://proxy.ccs.ornl.gov:3128"
+export http_proxy="http://proxy.ccs.ornl.gov:3128"
+export https_proxy="http://proxy.ccs.ornl.gov:3128"
+export HTTP_PROXY="http://proxy.ccs.ornl.gov:3128"
+export HTTPS_PROXY="http://proxy.ccs.ornl.gov:3128"
+export proxy="proxy.ccs.ornl.gov:3128"
+export no_proxy='localhost,127.0.0.0/8,*.ccs.ornl.gov,*.olcf.ornl.gov,*.ncrc.gov'
+
+echo $PATH
+which unzip
+
+export MY_CLUSTER=ascent
+. buildsystem/spack/load_spack.sh &&
+spack develop --no-clone --path=$(pwd) exago@develop &&
+spack develop --clone --force FORCE --path=$(pwd)/hiop hiop@develop &&
+cd $(pwd)/hiop &&
+git submodule update --init --recursive &&
```

Comment:
Small script here ensures a fresh checkout of hiop@develop whenever pipelines are ran.

This might cause issues, but should be extended to other CI platforms in the future - see #10 or a new issue

Reply:
Since latest develop causes build failures, what I think I need to do is:
- cherry-pick commits from @nychiang 's PR in #30 
- rebuild hiop@develop on all platforms
- merge
- revisit previously failing tests

cc @abhyshr